modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/internal/asset/entry/provider/test/AssetListAssetEntryProviderImplTest.java

```diff
@@ -0,0 +1,269 @@
+	}
+
+	@Test
+	public void testCombineSegmentsOfDynamicCollection() throws Exception {
+		Dictionary<String, Object> dictionary =
+			HashMapDictionaryBuilder.<String, Object>put(
+				"enabled", true
+			).build();
+
+		try (ConfigurationTemporarySwapper configurationTemporarySwapper =
+				new ConfigurationTemporarySwapper(
+					"com.liferay.asset.list.web.internal.configuration." +
+						"FFCollectionsVariationsPrioritizationConfiguration",
+					dictionary)) {
+
+			AssetListEntry assetListEntry =
+				_assetListEntryLocalService.addAssetListEntry(
+					TestPropsValues.getUserId(), _group.getGroupId(),
+					"Dynamic title", AssetListEntryTypeConstants.TYPE_DYNAMIC,
+					_getTypeSettings(), _serviceContext);
+
+			SegmentsEntry segmentsEntry1 = _addSegmentsEntry(
+				_group.getGroupId(), TestPropsValues.getUser());
+
+			SegmentsEntry segmentsEntry2 = _addSegmentsEntry(
+				_group.getGroupId(), TestPropsValues.getUser());
+
+			long[] segmentsEntryIds = {
+				segmentsEntry1.getSegmentsEntryId(),
+				segmentsEntry2.getSegmentsEntryId()
+			};
+
+			AssetListEntrySegmentsEntryRel variation1 =
+				AssetListTestUtil.addAssetListEntrySegmentsEntryRel(
+					_group.getGroupId(), assetListEntry,
+					segmentsEntry1.getSegmentsEntryId());
+
+			AssetListEntrySegmentsEntryRel variation2 =
+				AssetListTestUtil.addAssetListEntrySegmentsEntryRel(
+					_group.getGroupId(), assetListEntry,
+					segmentsEntry2.getSegmentsEntryId());
+
+			variation1.setPriority(1);
+			variation2.setPriority(0);
+
+			_addJournalArticle(new long[0]);
+			_addJournalArticle(new long[0]);
+
+			JournalArticle journalArticle = _addJournalArticle(new long[0]);
+
+			List<AssetEntry> assetEntries =
+				_assetListAssetEntryProvider.getAssetEntries(
+					assetListEntry, segmentsEntryIds);
+
+			Assert.assertEquals(
+				assetEntries.toString(), 6, assetEntries.size());
+
+			AssetEntry firstAssetEntry = assetEntries.get(0);
+
+			Assert.assertEquals(
+				firstAssetEntry.getTitle(LocaleUtil.US),
+				journalArticle.getTitle(LocaleUtil.US));
+		}
+	}
+
+	private JournalArticle _addJournalArticle(long[] assetCategories)
+		throws Exception {
+
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext(
+				_group.getGroupId(), TestPropsValues.getUserId(),
+				assetCategories);
+
+		return JournalTestUtil.addArticle(
+			_group.getGroupId(),
+			JournalFolderConstants.DEFAULT_PARENT_FOLDER_ID, serviceContext);
+	}
+
+	private SegmentsEntry _addSegmentsEntry(long groupId, User user)
+		throws Exception {
+
+		Criteria criteria = new Criteria();
+
+		_segmentsCriteriaContributor.contribute(
+			criteria,
+			String.format("(firstName eq '%s')", user.getFirstName() + "a"),
+			Criteria.Conjunction.AND);
+
+		return SegmentsTestUtil.addSegmentsEntry(
+			groupId, CriteriaSerializer.serialize(criteria),
+			User.class.getName());
+	}
+
+	private String _getTypeSettings() {
+		return UnicodePropertiesBuilder.create(
```

Comment:
Do you need we need all this properties?

Reply:
Refactored!